advanced.php

```diff
@@ -158,17 +158,18 @@ function query_form($query="")
 
   if($_SERVER['REQUEST_METHOD'] == 'POST')
   {
+    validateToken();
```

Comment:
Each time a form is rendered the csrfToken() will create a new token, but when a post request is made it won't generate another one so it'll always be valid. I tried it on both php 5.6 and php 7. You're right about the repetitive code, I'll fix it on a future commit.

Reply:
I don't know exactly which scenario you refer to as "repetitive code" (is it just the many `if`s?) but I was under the impression that if I have multiple forms and multiple `csrfToken()`s then the HTML code would look like this:

```
<form id="form1">
<input type="hidden" name="token" value="1234">
</form>
<form id="form2">
<input type="hidden" name="token" value="5678">
</form>
<form id="form3">
<input type="hidden" name="token" value="90ab">
</form>
```

and if I submit `form1` or `form2`, then the token will be wrong, as only `90ab` is valid and stored in the `$_SESSION`. Is this not a problem? Maybe I didn't understand the code properly, but I have to admit that I have not tested that.

Reply:
Yes by repetitive code I mean that validateToken() could be called only once with the `if` implementation as you suggested. Each time the csrfToken() is called(every time a form is rendered) the $_SESSION['token'] changes to a random value `$token = $_SESSION['token'] = bin2hex(random_bytes(20));` so the validateToken() will check the newly created token rather than the old one.